src/services/layerHelper.js

```diff
@@ -0,0 +1,282 @@
+  var parser = new ol.format.WMTSCapabilities();
+  var layer = new ol.layer.Tile();
+  var deferred = this.q_.defer();
+
+  this.http_.get(capabilitiesUrl).then(function(response) {
+    var result;
+    if (response.data) {
+      result = parser.read(response.data);
+    }
+    if (goog.isDef(result)) {
+      var options = ol.source.WMTS.optionsFromCapabilities(result,
+          {layer: layerName, requestEncoding: 'REST'});
+      layer.setSource(new ol.source.WMTS(options));
+      deferred.resolve(layer);
+    } else {
+      deferred.resolve();
+    }
+  }, function(response) {
+    deferred.resolve();
+  });
+
+  return deferred.promise;
+};
+
+
+/**
+ * Create and return an ol.layer.Group. You can pass a collection of layers to
+ * directly add them in the returned group.
+ * @param {ol.Collection.<ol.layer.Base>=} opt_layers The layer to add to the
+ * returned Group.
+ * @return {ol.layer.Group}
+ * @export
+ */
+ngeo.LayerHelper.prototype.createBasicGroup = function(opt_layers) {
+  var group = new ol.layer.Group();
+  if (goog.isDefAndNotNull(opt_layers)) {
+    group.setLayers(opt_layers);
+  }
+  return group;
+};
+
+
+/**
+ * Get the position of the layer in the map's layers array or -1 if the layer
+ * can't be found.
+ * Experimental. Tested only with ol.layer.Image and ol.layer.Tile.
+ * @param {ol.layer.Layer} layer The concerned layer.
+ * @return {number} index or -1 if not found.
+ * @export
+ */
+ngeo.LayerHelper.prototype.getLayerIndex = function(layer) {
+  var i, n;
+  var layers = /** @type {Array.<ol.layer.Layer>} */
+      (this.map_.getLayers().getArray());
+  var layerName = this.getLayerName(layer);
+  if (goog.isDefAndNotNull(layerName)) {
+    for (i = 0; i < layers.length; i++) {
+      n = this.getLayerName(layers[i]);
+      if (n === layerName) {
+        return i;
+      }
+    }
+  }
+  return -1;
+};
+
+
+/**
+ * Retrieve the name of the given layer or null if no name is found.
+ * Experimental. Tested only with ol.layer.Image and ol.layer.Tile.
+ * @param {ol.layer.Layer} layer The concerned layer.
+ * @return {string} layer name or null;
+ * @export
+ */
+ngeo.LayerHelper.prototype.getLayerName = function(layer) {
+  var layerName, layerSource;
+  if (layer instanceof ol.layer.Image) {
+    layerSource = /** @type {ol.source.ImageWMS} */ (layer.getSource());
+    layerName = layerSource.getParams()['LAYERS'];
+  } else if (layer instanceof ol.layer.Tile) {
+    layerSource = layer.getSource();
+    if (layerSource instanceof ol.source.WMTS) {
+      layerName = layerSource.getLayer();
+    }
+  }
+  return layerName || null;
+};
+
+
+/**
+ * Add the given layer on the map if it doesn't already exists and if the layer
+ * has a source.
+ * Experimental. Tested only with ol.layer.Image and ol.layer.Tile.
+ * @param {ol.layer.Layer} layer The concerned layer.
+ * @return {boolean} true if added, false else.
+ * @export
+ */
+ngeo.LayerHelper.prototype.addLayerToMap = function(layer) {
```

Comment:
Yes. Do you ask that because of the ``...if the layer has a source.`` ?
In my gmf-layertree, you can still have WMTS without source (on WMTS source errors). So a check before to add a layer on the map is useful. And this function doesn't allow to add a same layer twice (that is useful too, so I don't need to do that when I add a group of layers).